rspec.md

```diff
@@ -0,0 +1,87 @@
+# RSpec Coding Style
+
+## Contents
+
+* [What do we use RSpec for?](#what-do-we-use-rspec-for)
+* [Why RSpec?](#why-rspec)
+* [Example unit test](#example-unit-test)
+* [Example integration test](#example-integration-test)
+
+## What do we use RSpec for?
+
+We use RSpec to write both unit and integration tests for our Ruby code.
+
+## Why RSpec?
+
+[RSpec is the most popular test framework for Ruby][ruby-toolbox], and is [recommended for Rails][gorails].
+
+This explains why we use RSpec for our unit tests. For integration tests, there is another popular alternative to RSpec, and that is Cucumber.
+
+Many GOV.UK applications have some or all of their integration tests written in Cucumber. However, we are gradually moving away from Cucumber in favour of using RSpec, for reasons outlined in this article: [How we write readable feature tests with RSpec][chris-zetter-blogpost].
+
+Essentially, using RSpec for both BDD and TDD requires less of a context switch than using a separate tool for BDD.
+
+[ruby-toolbox]: https://www.ruby-toolbox.com/categories/testing_frameworks
+[gorails]: https://gorails.com/tool_categories/test-frameworks/tools
+[chris-zetter-blogpost]: https://about.futurelearn.com/blog/how-we-write-readable-feature-tests-with-rspec
+
+## Example unit test
+
+This example is taken from [finder-frontend][finder-frontend]:
+
+[finder-frontend]: https://github.com/alphagov/finder-frontend
+
+```ruby
+describe ContentItem do
```

Comment:
I would like to suggest we avoid the pollution of the global namespace that came with early RSpec and use `RSpec.describe`, `RSpec.feature` etc when opening top level tests like this.

We should probably be enforcing this by setting `config.expose_dsl_globally = false` in our test setup.

See https://relishapp.com/rspec/rspec-core/docs/configuration/global-namespace-dsl